### PR TITLE
Update README.md for installing dependencies before compiling

### PR DESCRIPTION
**Pull request by Soptq**
## Motivation

Some libraries are missing during compiling the `aptos` from the source code, and there is no instruction on how to install them. I think it would be helpful to mention it in the `README.md`

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

(You must have submitted a [signed CLA](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#contributor-license-agreement) that includes your GitHub handle prior to us accepting and landing your work. Write your answer here.)

Submitted to cla@aptoslabs.com

## Test Plan

No test plan for the documentation update.

## Related PRs

N/A

### CODE REVIEW

File: crates/aptos/README.md [hunk 1]
```diff
@@ -20,12 +20,16 @@ You will need the `cargo` package manager to install the `aptos` CLI tool.  Foll
 source $HOME/.cargo/env
 ```
 ### Install the `aptos` CLI
-1. Install the `aptos` CLI tool by running the below command.  You can run this command from any directory.  The `aptos`
+1. Install dependencies before compiling:
+   1. For Debian or Ubuntu: `sudo apt install build-essential pkg-config openssl libssl-dev libclang-dev`.
+   2. For RHEL or Centos: `sudo yum install pkgconfig openssl openssl-devel clang`.
+   3. For others: please manually install `pkg-config` `openssl`, `libssl` and `libclang`.
```
Comment (gregnazario) [inline crates/aptos/README.md:26]: You could probably also mention the `scripts/devsetup.sh` in the repo, but that might make it more complicated.  @rajkaramchedu is this clear enough?

**Comment by gregnazario** [comment]
Also, thanks for filling out the CLA ahead of time and thanks for updating this missing piece.  Sometimes I forget that not everyone has these

File: crates/aptos/README.md (hunk 1 above)
Comment (rajkaramchedu) [reply crates/aptos/README.md:26]: For step 3, can we add the command example to manually install? For example, we can say, "npm pkg-config" (if that is correct), then the user will understand it better.

Comment (rajkaramchedu) [reply crates/aptos/README.md:26]: Obviously "for others" cover a whole gamut but "npm" will give the user a clear idea what we mean.

**Review by rajkaramchedu — APPROVED** [review]
A couple comments. Approving so I am not blocking.

File: crates/aptos/README.md (hunk 1 above)
Comment (Soptq) [reply crates/aptos/README.md]: Updated, thanks for you advice!

**Comment by gregnazario** [comment]
@Soptq please update the PR to fix this lint error, and then we can push it through:
```
[ERROR] [trailing-whitespace] [content crates/aptos/README.md]: trailing whitespace at line 30

Error: there were lint errors
```

**Comment by Soptq** [comment]
> @Soptq please update the PR to fix this lint error, and then we can push it through:
> 
> ```
> [ERROR] [trailing-whitespace] [content crates/aptos/README.md]: trailing whitespace at line 30
> 
> Error: there were lint errors
> ```

Updated, thanks for notifying :)

**Pull request closed without merging**